HW2/P2/parallel_vector.pyx

```diff
@@ -107,9 +112,14 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
```

Comment:
what if counts[src[idx]] changes between now and acquiring the lock? You usually want to check the same condition again after acquiring the lock